examples/ghc-lib-test-mini-compile/src/Main.hs

```diff
@@ -394,16 +412,51 @@ fakeSettings = Settings {
 
     platform =  genericPlatform
 
+#elif (defined (GHC_9_4) || defined (GHC_9_6) || defined (GHC_9_8) || defined (GHC_9_10) || defined (GHC_9_12))
+    sGhcNameVersion=ghcNameVersion
+  , sFileSettings=fileSettings
+  , sTargetPlatform=platform
+  , sPlatformMisc=platformMisc
+  , sToolSettings=toolSettings
+  }
+  where
+    fileSettings = FileSettings {
```

Comment:
Since the only difference between GHC_9_2 and GHC_9_4 - GHC_9_12 and then GHC_9_14 is the FileSettings struct, can we maybe use a nested if statement to only change fileSettings between the two versions? e.g.

```haskell
#else
    {- defined (GHC_9_4) || defined (GHC_9_6) || defined (GHC_9_8) || defined (GHC_9_10) || defined (GHC_9_12) || defined (GHC_9_14) -}
    sGhcNameVersion=ghcNameVersion
    ...
    #if defined(GHC_9_14)
        sUnitSettings=unitSettings
    #endif
    fileSettings = FileSettings {
    #if defined (GHC_9_2)
        fileSettings_tmpDir="."
        ...
    #else
        fileSettings_topDir="."
        ...
    #endif
    
    toolSettings = ToolSettings {
    ...
#else
...
```

I see why this may look more confusing than just flattening all the cases, so it's up to you. The main advantage is that we only need to change a common line like `platformMisc = PlatformMisc {` in one place.